### PR TITLE
feat(deepnote-file): Migration to single notebook deepnote file

### PR DESCRIPTION
**Pull request by tkislan**
## Summary by CodeRabbit

* **New Features**
  * Added `deepnote split <path>` to generate one-notebook-per-file outputs (with `--output` and `--force`) and included shell completion.
  * Added `--outputFormat` to the convert CLI.
* **Bug Fixes**
  * Improved snapshot discovery/loading with optional notebook-scoped behavior to prevent cross-notebook mixing.
  * Enhanced sibling init composition and execution snapshot persistence across run/analyze/lint/stats.
  * Added a multi-notebook lint warning when multiple non-init notebooks are present.
* **Documentation**
  * Updated `.deepnote` organization guidance and snapshot/split naming documentation.
* **Tests**
  * Expanded coverage for split, snapshot lookup/splitting, and sibling init resolution/execution.

### CODE REVIEW

File: packages/convert/src/snapshot/resolve-init.ts [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+/**
+ * Compare two settings objects for structural equality via JSON serialization.
+ * Treats `undefined` and `{}` as equivalent.
+ */
+function areSettingsEqual(a: unknown, b: unknown): boolean {
+  const norm = (value: unknown): string => {
+    if (value === undefined || value === null) return 'null'
+    return JSON.stringify(value, sortedKeysReplacer)
+  }
+  return norm(a) === norm(b)
```
[the remark this answers is not in the document]
Comment (tkislan) [reply packages/convert/src/snapshot/resolve-init.ts:262]: Holding off on this one. The `{}` vs `undefined` mismatch only produces a
non-fatal, informational divergence warning — it doesn't affect init resolution
or execution. It also requires a specific shape: a split sibling whose `settings`
serializes to an empty object while the main file omits it entirely. In practice
both files go through the same serializer during this migration, so settings are
consistently either present or absent on both sides.
Since this is an ongoing migration to the single-notebook format where the
snapshot/settings plumbing is still moving, I'd rather not add normalization
special-cases that could mask a genuine divergence. Tracking it as a known minor
edge case and will revisit if false warnings actually show up.

**Comment by tkislan** [comment]
@coderabbitai resolve the review comment in run.test.ts file

Skipping for now. The underlying fix already landed — init scope and prelude
selection are keyed by `initNotebookId` everywhere (4d288b3), and the resolver
matches the init sibling by id, not name, which the existing resolver tests
already exercise.
Duplicate notebook names across sibling .deepnote files are an edge case rather
than a normal workflow. A dedicated duplicate-name fixture is reasonable
hardening, but while the run/preflight plumbing is still in flux during this
migration I'd prefer not to pin it down with a narrow regression test we may have
to rewrite. Noting as a nice-to-have follow-up.

**Review by dinohamzic — COMMENTED** [review]
AI review: can you please check if anything is relevant? I'll start manual testing in the meantime.

- **[P1] Unscoped loads can select a split sibling’s snapshot.** [lookup.ts:91](/Users/dino/Development/deepnote/packages/convert/src/snapshot/lookup.ts:91) includes notebook-scoped snapshots when no `notebookId` is supplied, then `loadLatestSnapshot` selects the first. The original multi-notebook file can therefore load one arbitrary split notebook’s outputs. Unscoped loads should use legacy snapshots only, or merge all relevant scoped snapshots.

- **[P2] Snapshots are treated as sibling init sources.** [resolve-init.ts:149](/Users/dino/Development/deepnote/packages/convert/src/snapshot/resolve-init.ts:149) accepts every `*.deepnote`, including `*.snapshot.deepnote`. If snapshots are stored beside sources, the init source plus its snapshot become two matches and execution fails as ambiguous. Exclude snapshot files.

- **[P2] Generated snapshot names do not always round-trip.** [lookup.ts:24](/Users/dino/Development/deepnote/packages/convert/src/snapshot/lookup.ts:24) rejects notebook IDs beginning with `_` or `-`, although the encoder preserves them. Additionally, [split.ts:93](/Users/dino/Development/deepnote/packages/convert/src/snapshot/split.ts:93) converts ISO timestamp colons to underscores, causing the parser to misinterpret part of the timestamp as the notebook ID.

- **[P2] Failed splits can leave partial output.** [split.ts:41](/Users/dino/Development/deepnote/packages/cli/src/commands/split.ts:41) writes sequentially. If a later target exists, earlier files remain created despite the command failing. Preflight all targets when `--force` is absent, or use temporary files and rename.

- **[P2] Required CLI reference documentation is missing.** The new command at [cli.ts:457](/Users/dino/Development/deepnote/packages/cli/src/cli.ts:457) is absent from [cli-utility.md](/Users/dino/Development/deepnote/skills/deepnote/references/cli-utility.md:1), contrary to the repository’s AGENTS.md synchronization requirement.

**Review by dinohamzic — COMMENTED** [review]
Repo README analysis

Key conflicts:

- Root [README.md](/Users/dino/Development/deepnote/README.md:30) explicitly encourages organizing multiple notebooks in one `.deepnote` file.
- [CLI README](/Users/dino/Development/deepnote/packages/cli/README.md:160) promotes “Run all notebooks,” omits the new `deepnote split` command, and documents directory conversion at [line 401](/Users/dino/Development/deepnote/packages/cli/README.md:401).
- [Convert README](/Users/dino/Development/deepnote/packages/convert/README.md:47) promotes converting directories of notebooks; the implementation still combines them into one multi-notebook `.deepnote`.
- [MCP README](/Users/dino/Development/deepnote/packages/mcp/README.md:100) describes running all notebooks without marking this as legacy compatibility.

Only [FILES.md](/Users/dino/Development/deepnote/FILES.md:26), [deepnote-format.md](/Users/dino/Development/deepnote/docs/deepnote-format.md:19), and the Deepnote skill were updated. The README coverage is therefore incomplete and partly contradictory.

**Review by dinohamzic — COMMENTED** [review]
Functionality wise this is working great, I was not able to find any issue while manually testing. 👏

File: packages/convert/README.md
Comment (dinohamzic) [inline packages/convert/README.md:1]: It seems this README is still referencing deprecated/removed APIs, can you please double check?

**Review by dinohamzic — COMMENTED** [review]
Some semi-minor AI findings + an additional comment

```
- [P2] Snapshot filenames do not round-trip: packages/convert/src/snapshot/lookup.ts:27 ambiguously parses notebook IDs containing an underscore-delimited UUID. ISO timestamps sanitized by generateSnapshotFilename also return
    null when parsed. Such snapshots become undiscoverable.
 
- [P2] Split can leave partial output: packages/cli/src/commands/split.ts:41 writes sequentially before discovering a later collision. Preflight all targets or stage writes before committing them.
 
- [P2] Directory conversion can create duplicate snapshot identities: packages/cli/src/utils/to-deepnote-conversion.ts:47 shares one project ID while packages/convert/src/jupyter-to-deepnote.ts:100 preserves notebook IDs. Copied
    notebooks with the same metadata therefore overwrite each other’s snapshots.

- [P2] Init-prelude documentation is inaccurate: skills/deepnote/references/cli-utility.md:25 says the sibling init runs as a prelude, but --notebook and --block execution filter it out. Qualify the docs or implement prelude
    behavior for filtered runs.
```

**Review by dinohamzic — COMMENTED** [review]
No issues found while testing manually. 👌

**Review by dinohamzic — COMMENTED** [review]
- **[P1] Split files lose init context in analysis.** [`split.ts`](/Users/dino/Development/deepnote/packages/convert/src/snapshot/split.ts:279) keeps only the main notebook, while lint/analyze/DAG load it without resolving the sibling init. Dependencies defined in init disappear from the DAG. Make analysis loaders init-aware, as run already is.

- **[P1] Copied Jupyter notebooks can overwrite each other's snapshots.** Directory conversion shares one project ID ([utility](/Users/dino/Development/deepnote/packages/cli/src/utils/to-deepnote-conversion.ts:47)) while preserving `deepnote_notebook_id` ([converter](/Users/dino/Development/deepnote/packages/convert/src/jupyter-to-deepnote.ts:100)). Duplicate IDs therefore produce identical snapshot paths. Detect and regenerate/reject duplicates.

- **[P2] Ordinary `.py` helpers abort directory conversion.** [`to-deepnote-conversion.ts`](/Users/dino/Development/deepnote/packages/cli/src/utils/to-deepnote-conversion.ts:19) calls throwing `detectFormat()` inside `filter()`. Mixed notebook/helper directories fail instead of ignoring unsupported Python files.

- **[P2] MCP drops sibling-init divergence warnings.** [`execution.ts`](/Users/dino/Development/deepnote/packages/mcp/src/tools/execution.ts:138) discards `resolved.warnings`, hiding settings/integration mismatches that CLI reports.

- **[P2] Documentation still uses removed APIs and legacy snapshot names.** [`converting-notebooks.md`](/Users/dino/Development/deepnote/docs/converting-notebooks.md:199) imports the removed plural converter; [`FILES.md`](/Users/dino/Development/deepnote/FILES.md:184) omits the new notebook-ID filename segment.

**Review by dinohamzic — APPROVED** [review]
